commons/src/main/java/com/powsybl/commons/reporter/Reporter.java

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright (c) 2021, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.commons.reporter;
+
+import java.util.Map;
+
+/**
+ * @author Florian Dupuy <florian.dupuy at rte-france.com>
+ */
+public interface Reporter {
+
+    Reporter NO_OP = new NoOpImpl();
+
+    Reporter createChild(String taskKey, String defaultName, Map<String, Object> values);
+
+    Reporter createChild(String taskKey, String defaultName);
+
+    Reporter createChild(String taskKey, String defaultName, String key, Object value);
+
+    void addTaskValue(String key, Object value);
+
+    void report(String reportKey, String defaultMessage, Map<String, Object> values);
```

Comment:
Do we want a system where the code doesn't create the report data if the severity is below a threshold in the context ? In slf4j they have info(data) and info(dataConsummer) for this